File: go/orm/sample/user_profile.go
package sample

import "github.com/kosmassp/SimpleOrm/go/orm"

// UserProfile is table user_profiles (STRICT) — the one-to-one fixture
// (ADR-0019 add.1): this side holds the foreign key with a unique index (the
// database is what makes a 1:1 a 1:1), so it declares an ordinary many-to-one;
// the inverse single navigation lives on User.
type UserProfile struct {
	BaseModel
	ID     int64 `orm:"column,key,generated"`
	UserID int64 `orm:"column"`
	// User is populated only by the library (Level 2 loading).
	User      *User   `orm:"many_to_one=UserID"`
	Bio       *string `orm:"column"`
	AvatarURL *string `orm:"column"`
}

func (UserProfile) Entity() orm.EntityDef {
	return orm.EntityDef{
		Source:      orm.Table("user_profiles"),
		ForeignKeys: []orm.ForeignKeyDef{orm.ForeignKey[User]("UserID")},
		Indexes:     []orm.IndexDef{orm.Index("UserID").Unique()},
	}
}

File: php/tests/Sample/Models/UserProfile.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Tests\Sample\Models;

use SimpleOrm\Metadata\Attributes\Column;
use SimpleOrm\Metadata\Attributes\ForeignKey;
use SimpleOrm\Metadata\Attributes\Generated;
use SimpleOrm\Metadata\Attributes\Index;
use SimpleOrm\Metadata\Attributes\Key;
use SimpleOrm\Metadata\Attributes\ManyToOne;
use SimpleOrm\Metadata\Attributes\Table;

/** Table `user_profiles` — the one-to-one fixture (ADR-0019 add.1): this side holds the FK with a unique index. */
#[Table('user_profiles')]
#[Index(['userId'], unique: true)]
final class UserProfile extends BaseModel
{
    #[Key]
    #[Generated]
    #[Column]
    public int $id;

    #[Column]
    #[ForeignKey(User::class)]
    public int $userId;

    #[ManyToOne('userId')]
    public private(set) ?User $user = null;

    #[Column]
    public ?string $bio = null;

    #[Column]
    public ?string $avatarUrl = null;
}
